dask/dataframe/_hyperloglog.py

```diff
@@ -0,0 +1,109 @@
+# -*- coding: utf-8 -*
+u"""Implementation of HyperLogLog
+
+This implements the HyperLogLog algorithm for cardinality estimation, found
+in
+
+    Philippe Flajolet, Éric Fusy, Olivier Gandouet and Frédéric Meunier.
+        "HyperLogLog: the analysis of a near-optimal cardinality estimation
+        algorithm". 2007 Conference on Analysis of Algorithms. Nice, France
+        (2007)
+
+"""
+from __future__ import division
+
+import numpy as np
+import pandas as pd
+
+def compute_first_bit(a):
+    # Consider making this less memory-hungry
+    #bits = np.bitwise_and.outer(a, 1<<np.arange(31, -1, -1))
+    bits = np.bitwise_and.outer(a, 1<<np.arange(32))
+    bits = bits.cumsum(axis=1).astype(np.bool)
+    return 33 - bits.sum(axis=1)
+
+def terrible_dataframe_hash(df):
```

Comment:
> Just for completeness (you might already know all this): Dask just uses an instance of multprocessing.Pool to do all the work. Prior to 3.4, anything that wasn't windows just used fork to spin up a process. After 3.4, the option to spawn processes with an entirely new interpreter state on non-windows was added to match the behavior of windows.....

I'm pretty sure this depends on the Manager you were using. If someone provided as context custom Pool with a distributed Manager, they would not just be dealing with forks. 

> `os.getenv('PYTHONHASHSEED') is None and (sys.platform == 'win32' or (hasattr(mp, 'get_start_method') and mp.get_start_method() == 'spawn'))`

BTW, I think this logic isn't quite right. (I certainly understand the idea you are conveying.)

> My preference is for the first one - I think it's reasonable to expect users to be able to set the hashseed, especially if the warning provided is detailed enough. I think the downsides of moving off of hash outweigh the benefits that something like hashlib might provide.

I can appreciate where you're coming from. FWIW, for the purposes of hyperloglog (which needs the whole shebang to look like random bits), `hash` isn't enough, but it could still be the main part of a solution by applying some fixed hash afterward.

> Argument might be that I haven't yet seen someone with "big data" dataframes that is using object dtype for anything but text. 

Yeah, same here. It's problematically hard, too, to get a good solution for a problem you've never actually seen, too. By supporting full, hetereogenous columns, there are some really subtle decisions to be made.